server/tilesource/cache.py

```diff
@@ -87,7 +87,8 @@ def __call__(cls, *args, **kwargs):
         instance = cache.get(key, _MARKER)
         if instance is _MARKER:
             instance = super(LruCacheMetaclass, cls).__call__(*args, **kwargs)
-            cache.put(key, instance)
+            if key is not None:
```

Comment:
I see now why you have this, but I think there's a cleaner way to support the functionality. See my comments above.